environments/centos7-base/main.py

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import stat
+import json
+import urllib2
+
+from datetime import datetime
+from subprocess import call
+from string import Template
+
+def error(message):
+    log(Template('ERR: $message').substitute(message=message))
+    sys.stdout.flush()
+
+def warn(message):
+    log(Template('WARN: $message').substitute(message=message))
+    sys.stdout.flush()
+
+def log(message):
+    print(Template('[main] $message').substitute(message=message))
+    sys.stdout.flush()
+
+# This function will attempt to read any results file, add our metadata to it,
```

Comment:
rather than putting comments before a function definition, python conventionally uses docstrings right at the start of the function body: https://www.python.org/dev/peps/pep-0257/